​[IMP] sales: updated margins for 19.4 (#18947)
odoo/documentation: content/applications/sales/sales/sales_quotations/margin.rst

Reviewer: ```suggestion
A *sales margin* is the profit gained from the sale of a product or service after all the costs
```

Reviewer: ```suggestion
In the Odoo **Sales** app, it is possible to show sales margins on quotations and sales
```
For whatever reason, I recall this being edited by multiple people many times in my own docs, so now I just use the shorthand as well.

Reviewer: ```suggestion
Each individual product must be configured to automatically calculate the sales margin on each quotation and sales order line. To enable this feature, navigate to
:menuselection:`Sales app --> Products --> Products`. Fill out the :guilabel:`Sales Price` and
:guilabel:`Cost` fields in the *General Information* tab for every desired product.
```
I think it's best to add the explanation in the beginning, since I went to the products and it was the Kanba view of all of them - it doens't explicitly say to click on the product until the very end. Also, tabs are italicized.

Reviewer: ```suggestion
Odoo calculates the margin with the following formula:
```
"by" seemed a bit off

Reviewer: ```suggestion
The margin percentage is calculated with the following formula:
```

Reviewer: ```suggestion
To calculate the margins on a sales order, navigate to :menuselection:`Sales app --> Orders --> Quotations` and click the :guilabel:`New` button to
create a new quotation. Fill out the quotation with the necessary information. While adding products
```
I would also add a link for the text "fill out the necessary information" to the doc or section of a doc that explains how to do that - might be helpful if they don't know.

Reviewer: This is a persnal preference, but I am not a fan of short numbered lists like this that have a lot of infomration but not a ton of explanations. I would recommend at a minimum, linking to the related docs to do whatever the ask is - like filling out the 'Create pricelist rules'. Also, a link to pricelist rules in general woudl help as well.

Author: I will keep the steps but I did follow your suggestion to add reference links.

Reviewer: This example is so long and scrolling, this can be a use case seciton instead - just a suggestion.

Baically add this instead of the admin block:

Use case: Conditional discounts
-------------------------------------------

A company selling denim clothing has created a seasonal discount where any cusotmer who purchases two or more pairs of denim clothing, between October and December, receive a 5% discount.

To enable this, first navigate to....

I owuld expand the details- because it just says 'it should look like this' - which is not enough detail.

Reviewer: ```suggestion
  app --> Orders --> Quotations`, then click either the :icon:`fa-area-chart` :guilabel:`(Graph)` icon or the
:icon:`oi-view-pivot` :guilabel:`(Pivot)` icon.  Next, click the :guilabel:`Measures` drop-down button and
deselect :guilabel:`Total` and select :guilabel:`Margin`. This presents all margin contributions across the customer base.
```

------------------------

To automatically calculate the sales margin for each quotation or sales order line item, go to
:menuselection:`Sales app --> Products --> Products`. Fill out the :guilabel:`Sales Price` and Each
individual product must be configured to automatically calculate the sales margin on each quotation
and sales order line. To enable this feature, navigate to :menuselection:`Sales app --> Products -->
Products`. Fill out the :guilabel:`Sales Price` and :guilabel:`Cost` fields in the *General

> Information* tab for every desired product.

Reviewer 2: This paragraph reads like there was a merge issue somewhere -- the "Fill out the Sales Price and..." sentence suddenly has a capitalized "Each" in it. I am assuming that somewhere in here, there was a comment and it merged weird. I would verify what you meant with this paragraph and make sure it's written out clearly.

.. image:: margin/sales-price-and-cost.png

>    :alt: Sale Price and Cost fields of a product form.

Reviewer 2: ```suggestion
   :alt: Sales Price and Cost fields of a product form.
```

Odoo calculates the margin with the following formula:

.. math::
   Sales~Margin = Sales~Price - Cost

The margin percentage is calculated with the following formula:

.. math::
   \frac{Sales~Price - Cost~Price}{Sales~Price} \times 100

Compute margins on sales orders
===============================

To calculate the margins on a sales order, navigate to :menuselection:`Sales app --> Orders -->
Quotations` and click the :guilabel:`New` button to create a new quotation. Fill out the quotation
with the necessary information. While adding products to the quotation, a new field,
:guilabel:`Margin`, automatically appears at the bottom of the to the quotation, a new
:guilabel:`Margin` field automatically appears at the bottom of the document. This field displays

> the order's total margin in the configured currency as well as the percentage.

Reviewer 2: This is the same issue where I think you merged something, and it ended up actually carrying it out weirdly. I would verify what you meant for this paragraph and clean it up accordingly.

.. image:: margin/sales-order-margin.png
   :alt: Sales order with Margin field on the bottom.

Compute margins on sales order lines
------------------------------------

To display a product's margin and the margin percentage per line item, click the
:icon:`oi-settings-adjust` :guilabel:`(settings adjust)` icon in the :guilabel:`Order Lines` tab.
Then, select the checkboxes for :guilabel:`Margin` and :guilabel:`Margin(%)`.

The :guilabel:`Margin` column shows the profit earned from the sale after accounting for all
associated costs, in the configured currency value. The :guilabel:`Margin(%)` shows the
margin value as a percentage. The margin for one unit is multiplied by the quantity to determine the
margin for the entire line.

The :guilabel:`Margin` and :guilabel:`Margin(%)` columns are not displayed by default, but once
enabled, the columns appear on all new and existing quotations and :abbr:`SOs (sales orders)`.
Editing the :guilabel:`Margin` column automatically recalculates the :guilabel:`Unit Price` on the
:abbr:`SO (sales order)`. Each column can be edited on the :abbr:`SO (sales order)` until an invoice
is created.

.. note::

>    The :guilabel:`Margin (%)` column becomes read only when the line item cost is `0`. To change the

Reviewer 2: ```suggestion
   The :guilabel:`Margin (%)` column becomes read-only when the line item cost is `0`. To change the
```

Adding a hyphen.

>    calculations for :guilabel:`Margin` and :guilabel:`Margin(%)` columns on the product form, refer

Reviewer 2: Is the guilabel Margin (%) with a space before the percent sign or without a space?

   to :ref:`Configure price and cost <sales_quotations/margin/configure-price-and-cost>` section for

>    more information.

Reviewer 2: ```suggestion
   to the :ref:`sales_quotations/margin/configure-price-and-cost` section for more information.
```

.. math::
   \frac{Sales~Price - Cost~Price}{Sales~Price} \times 100

.. image:: margin/sales-price-and-cost.png

>    :alt: Sale Price and Cost fields of a product form.

Reviewer: Normally I'd suggest all images go  at the bottom of a section, but I think in this case, it woudl be best tpo ave this above the calculations since that is what the chunk of text is referring to - and the calculations can be below (totally optional)

> document. This field displays the order's total margin in the configured currency as well as the

Reviewer: ```suggestion
to the quotation, a new  :guilabel:`Margin` field automatically appears at the bottom of the
document. This field displays the order's total margin in the configured currency as well as the
```

percentage.

> .. image:: margin/sales-order-margin.png

Reviewer: SUPER nitpick but there's a red splash on the upper-left corner - can you erase that? It just looks off.

   :alt: Sales order with Margin field on the bottom.

> To display a product's margin and the margin percentage per line item, click the
> :icon:`oi-settings-adjust` :guilabel:`(settings adjust)` icon in the :guilabel:`Order Lines` tab.

Reviewer: ```suggestion
Compiute margins on sales order lines
----------------------------------------------------

To display a product's margin and the margin percentage per line item, click the
:icon:`oi-settings-adjust` :guilabel:`(settings adjust)` icon in the :guilabel:`Order Lines` tab.
```
I don't know what this sould be called, but this can be a sub-section from above, sinc it is a slightly separate thing. The first half is the total marign, this is the individual margin.

Then, select the checkboxes for :guilabel:`Margin` and :guilabel:`Margin(%)`.

The :guilabel:`Margin` column shows the profit to earn from the sale after accounting for all
associated costs. It displays in the configured currency value. The :guilabel:`Margin(%)` shows the

> margin value as a percentage. The margin for one unit is multiplied by the quantity to determine the

Reviewer: ```suggestion
The :guilabel:`Margin` column shows the profit earned from the sale after accounting for all
associated costs, in the configured currency value. The :guilabel:`Margin(%)` shows the
margin value as a percentage. The margin for one unit is multiplied by the quantity to determine the
```

margin for the entire line.

The :guilabel:`Margin` and :guilabel:`Margin(%)` columns are not displayed by default, but once
enabled, the columns appear on all new and existing quotations and :abbr:`SOs (sales orders)`.
Editing the :guilabel:`Margin` column automatically recalculates the :guilabel:`Unit Price` on the
:abbr:`SO (sales order)`. Each column can be edited on the :abbr:`SO (sales order)` until an invoice
is created.

.. image:: margin/margin-margin-percent-checkboxes.png

>    :alt: Sales order's settings adjust menu with Margin and Margin(%) columns displayed.

Reviewer: This is so skinny and odd-looking, I'd remove it. Plus we go over how to use that menu in other basic docs, so it's not completely necessary (I think).

First go to :menuselection:`Sales app --> Products --> Pricelists` and create a new pricelist for
the seasonal discount. Click :guilabel:`Add a line` to configure a price rule for seasonal discount.
Include the specific product it applies to, the type of discount, the minimum quantity needed to
trigger the discount, and the date range the discount is active.

.. image:: margin/pricelist-configuration.png
   :alt: Pricelist Rules pop-up window.

After saving the pricelist, go to a :abbr:`SO (sales order)` containing blue denim jeans and select
the newly created pricelist in the :guilabel:`Pricelist` field.

.. image:: margin/sales-order-with-update-prices.png
   :alt: Sales order with a new pricelist selected.

.. important::
   The discount only applies if the :abbr:`SO (sales order)` meets both the date range and minimum
   quantity requirements.

After the changes are made, click :icon:`fa-refresh` :guilabel:`Update Prices` to update the
:guilabel:`Margin`, :guilabel:`Margin (%)`, and :guilabel:`Amount`. The margin is recalculated based
on the pricelist-adjusted product's sales price and cost.

.. image:: margin/sales-order-with-pricelist.png

>    :alt: Sales order with margins recalculated based on the price-list adjustment.

Reviewer 2: ```suggestion
   :alt: Sales order with margins recalculated based on the pricelist adjustment.
```

.. tip::
   Another way to visualize the impact of margins on sales orders is view the quotations data as a
   graph or chart. Go to :menuselection:`Sales app --> Orders --> Quotations`, then click either the
   :icon:`fa-area-chart` :guilabel:`(Graph)` icon or the :icon:`oi-view-pivot` :guilabel:`(Pivot)`
   icon. Next, click the :guilabel:`Measures` drop-down button and deselect :guilabel:`Total` and

>    select :guilabel:`Margin`. This presents all margin contributions across the customer base.

Reviewer 2: ```suggestion
   Another way to visualize the impact of margins on sales orders is to view the quotations data as
   a graph or chart. Go to :menuselection:`Sales app --> Orders --> Quotations`, then click either
   the :icon:`fa-area-chart` :guilabel:`(Graph)` icon or the :icon:`oi-view-pivot`
   :guilabel:`(Pivot)` icon. Next, click the :guilabel:`Measures` drop-down button and deselect
   :guilabel:`Total` and select :guilabel:`Margin`. This presents all margin contributions across
   the customer base.
```

You were missing a "to"; I fixed that and the paragraph line breaks for you here.

> 

Reviewer 2: Files only require one blank line at the end of it. You can get rid of this line so that there are no longer two blank lines at the end.
